californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -234,6 +235,13 @@ public Exchange receiveResponse(final Response response, final CorrelationContex
 			return null;
 		} else if (correlationContextMatcher.isResponseRelatedToRequest(exchange.getCorrelationContext(), responseContext)) {
 
+			if (response.getType() == Type.ACK && exchange.getCurrentRequest().getMID() != response.getMID()) {
```

Comment:
**NO!**

For separate responses the MID is not used for matching request/response!